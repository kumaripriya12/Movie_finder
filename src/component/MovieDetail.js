import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import axios from 'axios';
import {withRouter} from 'react-router-dom';

class MovieDetail extends Component {

	constructor(props){
		super(props);

		this.state = {
			movie_detail : null
		};

		this.goBack = this.goBack.bind(this);
	}

	componentDidMount() {
		try {
		//call to api using axios
	      axios.get('http://www.omdbapi.com/?apikey=b9bd48a6&i='+this.props.location.state.id)
		  .then((response) => {
		    this.setState({
		    	movie_detail : response.data 
		    });		    
		  });
        } catch(e) {
            console.log(e);
        }
	}

	goBack(){
		this.props.history.push({
			pathname : '/',
	        state :{
	        search_result : this.props.location.state.result
	        }
		});
	}

	render(){
			if(!this.state.movie_detail)
				return null;

			return(
				<div className="container pt-5">
					<div className="row">
						<div className="col-md-6">
							<img className="img-fluid img-thumbnail" src={this.state.movie_detail.Poster} alt={this.state.movie_detail.Title} style={{width: 350+'px'}} height='250px' />
						</div>
						<div className="col-md-6 text-left">
							<h1 className="font-weight-bold">{this.state.movie_detail.Title}</h1>
							<hr />
							<strong>{this.state.movie_detail.Plot}</strong>
							<hr />
							Starring - {this.state.movie_detail.Actors}
							<hr />
							Genre - {this.state.movie_detail.Genre}
							<hr />
							<small>Released on - {this.state.movie_detail.Released}</small>
						</div>
					</div>

					<div className="row">
						<div className="col text-center">
							<button className="btn btn-outline-dark btn-block" onClick={this.goBack}>Back</button>
						</div>
					</div>					
				</div>
			)
	}

}

export default withRouter(MovieDetail);