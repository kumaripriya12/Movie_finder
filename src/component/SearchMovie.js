import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { Redirect } from 'react-router';
import axios from 'axios';

import MovieDetail from './MovieDetail'

class SearchMovie extends Component {

	constructor(){
		super();

		this.state = {
			movies_result : null,
			selected_movieid : null,
			label_text : null
		};

		this.handleKeyUp = this.handleKeyUp.bind(this);
		this.handleClick = this.handleClick.bind(this);
	}

	componentDidMount(){
		if(this.props.location && this.props.location.state){
			this.setState({
				movies_result: this.props.location.state.search_result
			});

			//replace the history with null so that user can search for new movies & will not get stuck to previous search result
			this.props.history.replace('', null);
		}
	}

	handleKeyUp(event){
		

		let inputVal = event.target.value.trim(); //trim whitespaces
		let inputLen = inputVal.length; //gives the length of the input characters
		if(inputLen >= 2){
			
			try {
			//call to api using axios
		      axios.get('http://www.omdbapi.com/?apikey=b9bd48a6&s='+inputVal+'&type=movie')
			  .then((response) => {
			  	console.log("result", response.data);
			  	if(response.data.Response == "False"){
			  		this.setState({
				    	label_text : "No results found"
				    });
			  	}
			  	else {
			  		this.setState({
				    	label_text : null
				    });
			  	}
			    this.setState({
			    	movies_result : response.data.Search
			    });			    
			  });
	        } catch(e) {
	            console.log(e);
	        }
		}
	}	

	handleClick(e, id){
		if(id != null){
			this.setState({
				selected_movieid : id
			});
		}	
	}

	render(){
		if(this.state.selected_movieid){
			return(
				<Redirect to={{pathname: "/movie_detail", state: {id:  this.state.selected_movieid, result: this.state.movies_result}}} />
			)
		}

		return(
			<div className="container pt-5">
				<input className="border-secondary p-2" style={{width: '100%'}}  type="text" placeholder="Search Movies.." onKeyUp = {this.handleKeyUp} />
				<label className="text-danger">{this.state.label_text}</label>
				{
					this.state.movies_result ?
					<div>
						<ul className="list-group pt-2">
						    {
						      this.state.movies_result.map((item, i) => <li className="list-group-item" key={i} onClick={(e) => this.handleClick(e, item.imdbID)}>{item.Title}</li>)
						    }
						</ul>
					</div>
					: null
				}
			</div>
		)		
	}

}

export default SearchMovie;