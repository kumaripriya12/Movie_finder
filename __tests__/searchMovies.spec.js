import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure, mount, unmount } from 'enzyme';
import { MemoryRouter, Route } from "react-router-dom";

import SearchMovie from 'component/SearchMovie';
import MovieDetail from 'component/MovieDetail';

configure({adapter: new Adapter()});

    describe('Search movie component', ()=> {
        const wrapper = shallow(<SearchMovie />);

        it('should render', ()=> {
        	//component should have render
        	expect(wrapper.exists()).toBe(true);
        });

        it('should have a search button', ()=> {

            //There should be only one input
            expect(wrapper.find('input')).toHaveLength(1);
        });	

        it('should not break with articles', ()=> {
        	expect(wrapper.find("li")).toHaveLength(0);
        });	

        it('should find movie name', ()=> {
        	// wrapper.find('input').simulate('change', {target: {value: 'marvel'}});
        	// expect(wrapper.find("ul")).toHaveLength(1);

			const component = mount(<SearchMovie />);
			component.find('input').simulate('keyup', { target: {value: 'marvel'} });
			expect(component).toMatchSnapshot();
			component.unmount();

        });

        it("redirects / to /movie_detail", () => {
		  const wrapper = mount(
		    <MemoryRouter initialEntries={['/']}>
		      <Route component={MovieDetail} />
		    </MemoryRouter>
		  );

		  expect(wrapper.find(MovieDetail).length).toBe(1);
		});
        
        it("Movie detail render", () => {

          const params = {
          		state : {
          			id : 'tt4154664'
          		}
          }	

		  const wrapper = mount(
		    <MemoryRouter initialEntries={['/']}>
		      <Route component={MovieDetail} {...params}/>
		    </MemoryRouter>
		  );

		  expect(wrapper.find(MovieDetail).length).toBe(1);
		});

    });