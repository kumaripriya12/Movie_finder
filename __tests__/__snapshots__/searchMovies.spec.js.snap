// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search movie component should find movie name 1`] = `ReactWrapper {}`;
